### PR TITLE
Helped make this ready for contribution to MELPA

### PR DESCRIPTION
**Pull request by alexispurslane**
- Added a continuous integration system to automatically check that the package byte-compiles and meets MELPA's linting requirements on a few of the most recent Emacs versions.
- Went through and fixed all the lints MELPA considered blockers (and as many of the ones it didn't consider blockers as seemed reasonable)

Also, a few bug testing notes:

- Icons don't show up for me
- Groups don't work for me

and one feature suggestion:

- since you already have the description metadata for results from DBus, it would be really cool if you could add a consult annotation with it. I briefly looked into this, but I'm supposed to be taking a break from Emacs stuff right now, so I didn't go too deeply into it once I saw it would require a bit more plumbing than was easy (basically you'd have to add a separate function, with the keywords `:annotate` to the `consult--read` call, that could look at consult results and get the annotations you wanted out of them, probably from metadata attached to the strings, which I think you already do?)

### CODE REVIEW

File: consult-gnome-search.el [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ;; Copyright (C) 2024  Jürgen Hötzel, Alexis Purslane
 
 ;; Author: Jürgen Hötzel <juergen@hoetzel.info>, Alexis Purslane <alexispurslane@pm.me>
+;; URL: https://github.com/alexispurslane/consult-gnome-search
```
Comment (juergenhoetzel) [inline consult-gnome-search.el]: You change the URL of the package here?

**Comment by juergenhoetzel** [comment]
Thank you very much for your PR: Would you mind using the default Emacs settings for indentation? 

I just added a corresponding .dir-local settings in https://github.com/juergenhoetzel/consult-gnome-search/commit/047f00cb12760d73f98ebb13331ff83f12b7c55c

**Comment by juergenhoetzel** [comment]
> * Icons don't show up for me

The icon generation depends on your compiled-in Emacs image support. My Setup:

```elisp
image-types
(svg webp png gif tiff jpeg xpm xbm pbm)
```

File: consult-gnome-search.el (hunk 1 above)
Comment (alexispurslane) [reply consult-gnome-search.el]: Whoops, didn't mean to do that

**Comment by alexispurslane** [comment]
@juergenhoetzel 

> The icon generation depends on your compiled-in Emacs image support.

I have all the image support compiled in, the issue seems to be that it's just getting `nil` for any icon from DBus.

**Comment by juergenhoetzel** [comment]
@alexispurslane  Thanks for your contribution!

I have merged your changes (still had tab issues) and added fixes for the remaining lint and compilation warnings in PR #4

**Comment by alexispurslane** [comment]
Awesome! I'm not precisely sure where the tab issues are coming from, maybe I didn't reformat all the files correctly. Unfortunately I'm somewhat constrained on time.
— alexis (she/her)
Sent from Proton Mail Android
-------- Original Message --------On 6/19/24 4:31 PM, Jürgen Hötzel  wrote:
@alexispurslane  Thanks for your contribution!
I have merged your changes (still had tab issues) and added fixes for the remaining lint and compilation warnings in PR #4

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/juergenhoetzel/consult-gnome-search/pull/3#issuecomment-2179423027",
"url": "https://github.com/juergenhoetzel/consult-gnome-search/pull/3#issuecomment-2179423027",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Pull request closed without merging**